feat: add fixer for use-lifecycle-interface

rafaelss95 opened the pull request:
This PR aims to add a fixer for `use-lifecycle-interface`.

**Before**:
```ts
class UserEffects {
  ngrxOnInitEffects() {}
}
```

**After (--fix)**:
```ts
import { OnInitEffects } from '@ngrx/effects';
class UserEffects implements OnInitEffects {
  ngrxOnInitEffects() {}
}
```

rafaelss95 commented on src/utils/helper-functions/utils.ts:
```diff
@@ -10,3 +10,21 @@ export function findClassDeclarationNode(
+    ) as TSESTree.ImportDeclaration | undefined
+  }
+
+  // TODO: Remove this when enable tsConfig strict option.
+  return parentNode as TSESTree.ImportDeclaration | undefined
```
When I was implementing this, I noticed that `strict` is not enabled. I could send a follow-up PR if desired.

timdeschryver replied:
Good catch! Feel free to enable strict mode

rafaelss95 replied:
Done, #70.

Btw, thanks for adding me as a collaborator. It's a pleasure to help in this project :)

timdeschryver replied:
We have to thank you 🤗

timdeschryver commented on src/utils/helper-functions/guards.ts:
```diff
@@ -31,3 +31,15 @@ export function isClassDeclaration(
+export function isImportDeclaration(
+  node: TSESTree.Node,
+): node is TSESTree.ImportDeclaration {
+  return node.type === 'ImportDeclaration'
```
we should also check if `node` isn't `undefined` so it behaves the same as the other guards.

rafaelss95 replied:
Let's discuss it in #70.

timdeschryver commented on src/utils/helper-functions/guards.ts:
```diff
@@ -31,3 +31,15 @@ export function isClassDeclaration(
+export function isImportSpecifier(
+  node: TSESTree.Node,
+): node is TSESTree.ImportSpecifier {
+  return node.type === 'ImportSpecifier'
```
Same here

timdeschryver replied:
Same here

timdeschryver commented on src/utils/helper-functions/guards.ts:
```diff
@@ -31,3 +31,15 @@ export function isClassDeclaration(
+export function isImportDeclaration(
+  node: TSESTree.Node,
+): node is TSESTree.ImportDeclaration {
+  return node.type === 'ImportDeclaration'
```
We should also check if `node` is `undefined`, this way it behaves like the other guards

timdeschryver commented on src/rules/use-lifecycle-interface.ts:
```diff
@@ -36,27 +45,44 @@ export default ESLintUtils.RuleCreator(docsUrl)<Options, MessageIds>({
-            messageId,
-            data: {
-              methodName,
+        const {
```
Can we move this to inside the fixer scope?
This way this only gets executed when someone wants to fix the the code.

rafaelss95 replied:
Nice catch. I'll update it once #70 gets merged.

rafaelss95 replied:
Done!

timdeschryver reviewed (commented):
This is great, thanks @rafaelss95 !
I think this is our first rule with a fixer, do you think we should add a note to the docs?

rafaelss95 commented:
Hey @timdeschryver, I've addressed the feedback (code part)... about the docs part, I was thinking in something like https://github.com/cartant/eslint-plugin-rxjs#rules, WDYT?

timdeschryver commented:
@rafaelss95 yea, I was thinking the same thing :) 
I'll update the docs later, because I also might tweak the error level of the rules because most of them are set to `error` now which might be a bit too opionated :D
